core-bundle/src/Session/SessionFactory.php

```diff
@@ -12,25 +12,43 @@
 
 namespace Contao\CoreBundle\Session;
 
-use Symfony\Component\HttpFoundation\Session\SessionBagInterface;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\Session\Attribute\ArrayAttributeBag;
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpFoundation\Session\SessionFactoryInterface;
 use Symfony\Component\HttpFoundation\Session\SessionInterface;
 
 class SessionFactory implements SessionFactoryInterface
 {
+    public const SESSION_BAGS = [
+        'contao_backend' => '_contao_be_attributes',
+        'contao_frontend' => '_contao_fe_attributes',
+    ];
+
     public function __construct(
         readonly private SessionFactoryInterface $inner,
-        readonly private SessionBagInterface $backendBag,
-        readonly private SessionBagInterface $frontendBag,
+        readonly private RequestStack $requestStack,
+        readonly private ScopeMatcher $scopeMatcher,
     ) {
     }
 
     public function createSession(): SessionInterface
     {
         $session = $this->inner->createSession();
+        $request = $this->requestStack->getMainRequest();
```

Comment:
Are we sure that this should not be `getCurrentRequest()`? 😄

Reply:
I mean I suppose it's possible that a `?popup` _sub_-request can be made somewhere. But would that even trigger the session factory?

Reply:
No, I don‘t think so.